src/xrpld/app/misc/WasmHostFuncImpl.h

```diff
@@ -27,16 +27,29 @@ class WasmHostFunctionsImpl : public HostFunctions
 {
     ApplyContext& ctx;
     Keylet leKey;
+    std::shared_ptr<SLE const> currentLedgerObj;
 
     static int constexpr MAX_CACHE = 256;
     std::array<std::shared_ptr<SLE const>, MAX_CACHE> cache;
 
     void const* rt_ = nullptr;
 
+    Expected<std::shared_ptr<SLE const>, HostFunctionError>
+    getCurrentLedgerObj() const
+    {
+        if (currentLedgerObj)
+            return currentLedgerObj;
+        return Unexpected(HostFunctionError::LEDGER_OBJ_NOT_FOUND);
+    }
+
+    Expected<int32_t, HostFunctionError>
+    getActualCacheIndex(int32_t cacheIdx);
+
 public:
     WasmHostFunctionsImpl(ApplyContext& ctx, Keylet const& leKey)
         : ctx(ctx), leKey(leKey)
     {
+        currentLedgerObj = ctx.view().read(leKey);
```

Comment:
This should be inside of `getCurrentLedgerObj`

Reply:
I do it here because otherwise we'll keep checking for a bad object

Reply:
I mean  static initialization in getCurrentLedgerObj. We don't know if we will need it at all.

Reply:
This doesn't work unless I initialize it to `nullptr`:
```
src/xrpld/app/misc/WasmHostFuncImpl.h:42:30: error: no viable overloaded '='
   42 |             currentLedgerObj = ctx.view().read(leKey);
      |             ~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
```
class WasmHostFunctionsImpl : public HostFunctions
{
    ApplyContext& ctx;
    Keylet leKey;
    std::shared_ptr<SLE const> currentLedgerObj;
    bool objectCached = false;
 ...
 
 Expected<std::shared_ptr<SLE const>, HostFunctionError>
    getCurrentLedgerObj() 
    {
        if(!objectCached)
        {
            objectCached = true;
            currentLedgerObj = ctx.view().read(leKey);
        }
        if (currentLedgerObj)
            return currentLedgerObj;
        return Unexpected(HostFunctionError::LEDGER_OBJ_NOT_FOUND);
    }
...
}
```